toolkit/tools/internal/packagerepo/repocloner/rpmrepocloner/rpmrepocloner.go

```diff
@@ -324,8 +351,14 @@ func (r *RpmRepoCloner) WhatProvides(pkgVer *pkgjson.PackageVer) (packageNames [
 	}
 
 	foundPackages := make(map[string]bool)
+
+	effectiveCacheRepo := fetcherRepoID
```

Comment:
I would consider creating a simple function instead of this assignment to avoid potential mistakes in the future when someone might miss a place where `cacheRepoId` is calculated. I do not anticipate that issue to arise, but it might be a good idea to be defensive anyway. If you consider having a check for `IsRegularBuild` locally more readable, feel free to leave it as is.

Reply:
Added ` selectCorrectCacheRepoID()`